packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -553,15 +578,10 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
 
   async buildCustomTx(tx: BuildCustomTxInput): Promise<{ txToSign: SignTx<T> }> {
```

Comment:
nothing should be changing in this diff in

* `src/features/defi/providers/**/*`
* `src/lib/investor/**.*`

parts of the the source tree

changes to this function introduce risk and unnecessary changes to unrelated domains in the app

we should leave `buildCustomTx` alone, and follow the same wrapper pattern we did with `buildSendTx` vs `buildSignTx`, i.e.

copy paste the guts of `buildCustomTx`, call it `buildCustomApiTx`, lift the wallet dependency up, and have `buildCustomTx` inject the necessary bits  `buildCustomApiTx` needs to be pure, and new multi hop things and the api can use the new `buildCustomApiTx` wallet-free implementation

Reply:
gm
<img width="361" alt="image" src="https://github.com/shapeshift/web/assets/17035424/bd0122aa-64d1-4753-8d00-b8e3e0900a89">